dissect/target/plugins/os/windows/notifications.py

```diff
@@ -1,15 +1,160 @@
 import datetime
+from typing import Iterator, Optional
 from uuid import UUID
 
+from dissect.cstruct import cstruct
 from dissect.sql import sqlite3
 from dissect.util.ts import wintimestamp
 from flow.record import GroupedRecord
 
 from dissect.target.exceptions import UnsupportedPluginError
 from dissect.target.helpers.descriptor_extensions import UserRecordDescriptorExtension
-from dissect.target.helpers.record import create_extended_descriptor
+from dissect.target.helpers.record import WindowsUserRecord, create_extended_descriptor
 from dissect.target.plugin import Plugin, export
 
+appdb_def = """
+typedef struct {
+    char     Magic[4];                 // Always b"DNPW"
+    DWORD    Version;                  // Versions 1 (win8) and 3 (win10) seem to exist
+    QWORD    Timestamp;                // According to some sources this is LastNotificationDate,
+                                       // but that seems incorrect.
+    DWORD    NextNotificationId;
+    DWORD    Unknown;
+    char     Padding[8];
+} ChunkHeader;                         // size: 0x20
+
+typedef struct {
+    WORD     InUse;                    // ??
+    WORD     NextTileWriteIndex;       // ??
+    WORD     NextToastWriteIndex;      // ??
+    BYTE     Flags[2];
+} ChunkInfo;
+
+typedef struct {
+    QWORD    Timestamp1;               // ??
+    QWORD    Timestamp2;               // Is this time to refresh?
+    char     Uri[1024];                // Is this the correct size?
+    char     Padding[0x818 - 0x410];
+} PushDescriptor;                    // size: 0x818
```

Comment:
```suggestion
} PushDescriptor;                      // size: 0x818
```
I'm so sorry :smile:

Reply:
gah!